packages/core/utils/src/dml/helpers/create_mikro_orm_entity.ts

```diff
@@ -0,0 +1,93 @@
+import {
+  Entity,
+  OneToMany,
+  Property,
+  OneToOne,
+  ManyToMany,
+} from "@mikro-orm/core"
+import { DmlEntity } from "../entity"
+import { camelToSnakeCase, pluralize } from "../../common"
+import { upperCaseFirst } from "../../common/upper-case-first"
+import type { Infer, RelationshipType, SchemaType } from "../types"
+
+/**
+ * A helper function to define a Mikro ORM entity from a
+ * DML entity.
+ */
+export function createMikrORMEntity<
+  T extends DmlEntity<Record<string, SchemaType<any> | RelationshipType<any>>>
+>(entity: T): Infer<T> {
+  class MikroORMEntity {}
+
+  const className = upperCaseFirst(entity.name)
+  const tableName = pluralize(camelToSnakeCase(className))
+
+  /**
+   * Assigning name to the class constructor, so that it matches
+   * the DML entity name
+   */
+  Object.defineProperty(MikroORMEntity, "name", {
+    get: function () {
+      return className
+    },
+  })
+
+  Object.keys(entity.schema).forEach((property) => {
+    const field = entity.schema[property].parse(property)
```

Comment:
thought: I would expect our schemas to map to the column properties that we have used in the core. For example, the following are the columns for a `fulfillment_set` (simplified for brevity):
```json
{
  "id": {
    "name": "id",
    "type": "text",
    "unsigned": false,
    "autoincrement": false,
    "primary": false,
    "nullable": false,
    "mappedType": "text"
  },
  "name": {
    "name": "name",
    "type": "text",
    "unsigned": false,
    "autoincrement": false,
    "primary": false,
    "nullable": false,
    "mappedType": "text"
  },
  "metadata": {
    "name": "metadata",
    "type": "jsonb",
    "unsigned": false,
    "autoincrement": false,
    "primary": false,
    "nullable": true,
    "mappedType": "json"
  },
  "fulfillment_set_id": {
    "name": "fulfillment_set_id",
    "type": "text",
    "unsigned": false,
    "autoincrement": false,
    "primary": false,
    "nullable": false,
    "mappedType": "text"
  },
  "created_at": {
    "name": "created_at",
    "type": "timestamptz",
    "unsigned": false,
    "autoincrement": false,
    "primary": false,
    "nullable": false,
    "length": 6,
    "default": "now()",
    "mappedType": "datetime"
  },
},
```

So what I mean is we would have a mapping between our schema primitives and the MikroORM properties that look something along these lines:
- `text()` -> `"type": "text"` -> `@Property({ columnType: "text" })`
- `dateTime()` -> `"type": "timestamptz"` -> `@Property({ columnType: "timestamptz" })`
- `number()` -> `"type": "integer"` -> `@Property({ columnType: "integer" })`

And so on and so forth. 

What do you think? Also, let me know if I am missing something, I might not have the full context of this change.

Reply:
Maybe this is actually more about how the schema primitives define their `dataType` than this entity generation method. Because this approach here looks good to me.

Reply:
I am not sure I understand what you mean? there is still some changes to perform in regards to type vs columnType I believe as one is ts type and the other one is db type. maybe this is what you meant?

Reply:
Ah OK, maybe I am missing something. Does the `Property` on line 42-45 not correspond with the Property decorators we use on MikroORM models?

Reply:
it does, except that in our model when we apply the decorator it can infer the typescript type from the property, but with dynamic generation you dont have a typescript type and you have to set it manually. Here I think the usage must be changed as it is incorrect at the moment, but it is similar to what we have been doing with big number for example where we set both columnType and type, the first one for the db, the second one for the typing

Reply:
here is an example for dynamic apply of the decorator we used for big number
```
Property({
      type: "any",
      columnType: "numeric",
      trackChanges: false,
      ...options,
    })(target, columnName)
    ```

Reply:
if what you mean is the mapping is incorrect then yes it is, but I believe @thetutlage wanted to get early feedback on the approach mostly :)

Reply:
yes, as mentioned above, I believe virk has done anything around the mapping yet

Reply:
it is still a draft to validate the approach, i.e the way the class will be auto generated and the decorators applied, then the options of the decorators will be fixed 😄

Reply:
Got it, as long as we are aware of it, it's all good. Approach looks solid 💪

Reply:
yes ahah, I have a call with virk tomorrow morning, we will go through it 👍

Reply:
Yeah, have a call with @adrien2p tomorrow. My goal is to fill the gaps between our schema and MikroORM entities in that call.